auto sync 2022-03-09

nunnatsa opened the pull request:
auto sync upstream 2022-03-09

bardielle commented:
@nirarg @rgolangh it seems that there is a duplicate code (in providerid/controller.go) that look like ovirt providerid/controller.
Can we merge those two duplicates files?

bardielle commented:
/hold

rgolangh commented on pkg/providerid/controller.go:
```diff
@@ -59,12 +58,17 @@ func (r *providerIDReconciler) Reconcile(request reconcile.Request) (reconcile.R
 	}
 
-	node.Spec.ProviderID = FormatProviderID(r.infraClusterNamespace, id)
+	infraClusterNamespace, err := r.tenantClusterClient.GetNamespace()
```
infra ns = tenant ns ?

rgolangh commented on pkg/providerid/controller.go:
```diff
@@ -80,16 +84,20 @@ func FormatProviderID(namespace, name string) string {
 
 func (r *providerIDReconciler) getVMName(nodeName string) (string, error) {
-	vmi, err := r.infraClusterClient.GetVirtualMachineInstance(r.infraClusterNamespace, nodeName, &v1.GetOptions{})
+	infraClusterNamespace, err := r.tenantClusterClient.GetNamespace()
```
again mixing infra ns and tenant ns

nirarg replied:
**tenantClusterClient.GetNamespace** is function which return the namespace of the resources in the infracluster
This information is saved in the tenantcluster, in configMap
I would like to prevent from having long and complicated names (like GetInfraClusterResourcesNamespace)
Do you have any suggestions for other name?

bardielle replied:
@roy it's an issue that's not related to that PR 
@Nir open a jira ticket

nirarg replied:
Opened new jira task OCPCNV-159

rgolangh replied:
I don't get what's wrong with  `r.infraClusterNamespace`

rgolangh commented on pkg/managers/vm/machine_scope.go:
```diff
@@ -60,7 +60,7 @@ type machineScope struct {
 }
 
-func newMachineScope(machine *machinev1.Machine, tenantClusterClient tenantcluster.Client, infraClusterClientBuilder infracluster.ClientBuilderFuncType) (*machineScope, error) {
+func newMachineScope(machine *machinev1.Machine, tenantClusterClient tenantcluster.Client, infraClusterClient infracluster.Client) (*machineScope, error) {
```
to keep it consistent lets change the order of the args to be infraClusterClient, tenenatClusterClient

nirarg replied:
Fixed

bardielle commented:
/hold cancel

bardielle commented:
/approve

bardielle commented:
/lgtm

nirarg committed:
Change OWNERS file

kewang-equinix committed:
returen external IP address for loadbalancer

wangxin311 committed:
Update comments for timer

Wait 10 seconds for LoadBalancer to assign external IP after been created

nunnatsa committed:
Automatic sync with upstream

This PR adds a new github action to daily sync with upstream. It checks if there are new changes in upstream, and if there are, it creates a PR to sync.

The new PR is created by hard reseting to the upstream main branch, then restoring the downstream modified files.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
add script to sync with upstream

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
create PR

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

wangxin311 committed:
change from sleep timer to requeue

k8s-ci-robot committed:
Merge pull request #102 from wangxin311/lb

returen external IP address for loadbalancer

nirarg commented:
/lgtm
